### PR TITLE
Fix max and min with inf on CUDA

### LINKED ISSUE #10237: Irregular behavior of torch.max on cpu & gpu for a tensor containing -inf
## Issue description
```torch.max()``` has irregular behavior on a tensor containing all ```-inf```, depending on whether it is on cpu or gpu.
## Code example
```
>>> var = torch.zeros(5).log()
>>> var
-inf
-inf
-inf
-inf
-inf
[torch.FloatTensor of size 5]
>>> var.max()
-inf
# to gpu
>>> var = var.cuda()
>>> var
-inf
-inf
-inf
-inf
-inf
[torch.cuda.FloatTensor of size 5 (GPU 0)]
>>> var.max()
-3.4028234663852886e+38
```

## Environment
```
PyTorch version: 0.3.1
Is debug build: No
CUDA used to build PyTorch: 8.0.61

OS: CentOS Linux 7 (Core)
GCC version: (GCC) 4.8.5 20150623 (Red Hat 4.8.5-28)
CMake version: version 2.8.12.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX TITAN X
GPU 1: GeForce GTX TITAN X
GPU 2: GeForce GTX TITAN X
GPU 3: GeForce GTX TITAN X

Nvidia driver version: 390.48
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] msgpack-numpy (0.4.1)
[pip] numpy (1.13.1)
[pip] numpydoc (0.8.0)
[pip] torch (0.3.1)
[pip] torchtext (0.2.1)
[pip] torchvision (0.2.0)
[conda] Could not collect
```

### PR DESCRIPTION
**Pull request by ssnl**
Fixes #10237 #11084 

cc @vishwakftw

### CODE REVIEW

**Comment by vishwakftw** [comment]
@SsnL thank you for doing this. I am sorry for not being able to get to this.

File: test/test_torch.py [hunk 1]
```diff
@@ -793,9 +793,27 @@ def _testSelection(self, torchfn, mathfn):
     def test_max(self):
         self._testSelection(torch.max, max)
 
+    @staticmethod
+    def _test_max_with_inf(self, device='cpu'):
+        a = torch.tensor([[-inf, -inf, inf, 3], [inf, inf, -inf, -1]], device=device)
```
Comment (vishwakftw) [inline test/test_torch.py]: Is `inf` defined somewhere?

Comment (SsnL) [reply test/test_torch.py]: torch._six :) see the import at top of the file.

**Comment by SsnL** [comment]
@vishwakftw No need to be sorry. Thanks for all the good work you have been doing! :)

File: test/test_torch.py (hunk 1 above)
Comment (vishwakftw) [reply test/test_torch.py]: Oh oops, I missed that.

**Review by soumith — APPROVED** [review]

**Pull request closed without merging**